code4romania/de-urgenta-client
# Issue 25 add meta tags

## What?

- Added meta tags listed on #25 
![image](https://github.com/code4romania/de-urgenta-client/assets/93956840/4fd2cf1f-bc2b-4480-9b2d-b2555adfbd63)

- Added site icon & title (shown in browser tab)
![image](https://github.com/code4romania/de-urgenta-client/assets/93956840/ae08ee8b-42b0-4874-936b-4614811cc970)


closes #25

## How

- Adding meta tags which are plain text is trivial, configured those in the meta array found in `nuxt.config.js`
- From my research, the icon link and image tag should be static urls and the you [can't import images in nuxt.config.js](https://stackoverflow.com/questions/52683062/loading-ogimage-from-assets-in-nuxt-config-js)

To solve this:

1. I put the icon and site image in /static
2. Configured .env to hold the base url; the production env should contain this variable: `NUXT_BASE_URL="https://deurgenta.ro"`
3. Added the image meta tag and site icon inside the default layout

## Testing

To test meta tags locally you need a [local proxy](https://stackoverflow.com/questions/8569892/how-to-test-open-graph-on-localhost). I tried both [ngrok](https://ngrok.com) and [Localtunnel](https://localtunnel.github.io/www/) and they produced the same results for me.

To set up Localtunnel:

- Install it with `npm install -g localtunnel`
- Update the base url in .env to `NUXT_BASE_URL="https://de-urgenta.loca.lt`
- Run the project locally on port 3000 with `yarn dev`
- Start the proxy on that base url with `lt --port 3000 --subdomain de-urgenta`

Now you should be able to access the site on https://de-urgenta.loca.lt for a limited time and you can use that link on meta testing tools like https://metatags.io/

As you can see in the first screenshot when inspecting the page the tags are populated correctly, and pasting https://de-urgenta.loca.lt/image-de-urgenta.png in a browser lets me access the image with no issue. 

However, when testing on metatags.io the site image isn't being picked up
![image](https://github.com/code4romania/de-urgenta-client/assets/93956840/9b3c7cab-4d13-40bf-baad-f83fe73ff257)
nor is the icon when sending the link on slack
![image](https://github.com/code4romania/de-urgenta-client/assets/93956840/e781e88e-4bb7-47b5-9410-c5899898ba1b)

The code and HTML output looks good to me, so maybe it's just an issue with those proxys, but additional investigation would be welcomed (or at least re-test it after it makes it into prod and it's hosted for real)
